### DISCUSSION TITLE
The executable file freezed by v5.0 is twice larger than v4.10.

### DISCUSSION THREAD

**Discussion by rtrobin** [pyinstaller/pyinstaller#6755] [PyInstaller]
Glad to see that pyinstaller release v5.0. Tried freezing a small application into a single exe file with latest version. Turns out the executable file freezed by v5.0 is twice larger than v4.10 did.

Is this common? If so, why it is?

**Comment by rokm** [comment]
Can you provide an example or at least the list of packages that the application uses?

**Reply by rtrobin** [reply]
Here is the code I use. It helps me keep pc awake. Basically it press 'ctrl' button every minute. 

The package in use is [PySimpleGUI](https://github.com/PySimpleGUI/PySimpleGUI) and [PyAutoGUI](https://github.com/asweigart/pyautogui).

The package command I use is `pyinstall -wF main.py`

``` python
# main.py
import PySimpleGUI as sg
import pyautogui

menu_def = ['BLANK', ['!self use', '---', 'E&xit']]
tray = sg.SystemTray(menu=menu_def, data_base64=sg.DEFAULT_BASE64_ICON)

while True:  # The event loop
    menu_item = tray.read(timeout=1000*60)

    if menu_item == 'Exit':
        break
    elif menu_item == sg.EVENT_SYSTEM_TRAY_ICON_DOUBLE_CLICKED:
        break
    
    pyautogui.press('ctrl')

tray.close()
```

**Reply by bwoodsend** [reply]
I get 20MB in a clean Python 3.9 environment on Linux irregardless of which PyInstaller version I use.

**Reply by rtrobin** [reply]
Thanks for the reminder! I make two clean environment on windows to test. The size of freezed execution file is **10.0MB**(pyinstaller==4.10) vs **10.1MB**(pyinstaller==5.0).

After I freezed the application some days before, I have installed some other packages in my main environment. Pyinstaller seems to package something else into the file. In my main environment, the output package file is 30.0MB now.

**Reply by bwoodsend** [reply]
Yeah... PyInstaller's _automatic_ dependency tends to grab anything and everything it can see so you need to manually manage the dependencies PyInstaller can see then watch PyInstaller bounce aimlessly around your already correctly configured environment :upside_down_face:

**Reply by rtrobin** [reply]
It solves my problem. Though I can't mark your reply as an answer. Thanks.
